misc/async_matrix.py

```diff
@@ -0,0 +1,120 @@
+#!/usr/bin/env python3
+"""Test various combinations of generators/coroutines.
+
+This was used to cross-check the errors in the test case
+testFullCoroutineMatrix in test-data/unit/check-async-await.test.
+"""
```

Comment:
I figure this file doesn't need detailed review, so I'm not going to look at it carefully.